### ISSUE TITLE
Mouse release event ignored if released too quickly

### ISSUE THREAD

**Issue by aswerw** [EpicGames/raddebugger#91]
If you depress the left mouse button, and release it too quickly, the release event seems to be ignored. 

For example, if you do it over a line in the text view widget it'll be stuck selecting text afterwards, but it also happens with other widgets (menu items and tabs).

Here is a recording of it happening, by sending automated down and up events without any delay to reproduce it consistently:

https://github.com/EpicGames/raddebugger/assets/10407330/040eca64-5642-4f93-96db-92c17c0e3d29

**Comment by ryanfleury** [comment]
Fixed in fd0feefcf280538dd5b2c84885b97295f950ad57.

**Comment by aswerw** [comment]
This was initially fixed for me, but I think it broke again in 0d1ccd7.

I think the issue is with this if statement: 
```
// src\raddbg\raddbg.c
for(OS_Event *ev = leftover_events.first, *next = 0; ev != 0; ev = next)
{
  next = ev->next;
  if(ev->timestamp_us+1000000 < os_now_microseconds() ||
     ev->kind == OS_EventKind_Text ||
      (ev->kind == OS_EventKind_Press && ev->key != OS_Key_LeftMouseButton) ||
      (ev->kind == OS_EventKind_Press && ev->key != OS_Key_RightMouseButton) ||
      (ev->kind == OS_EventKind_Press && ev->key != OS_Key_MiddleMouseButton) ||
      (ev->kind == OS_EventKind_Release && ev->key != OS_Key_LeftMouseButton) ||
      (ev->kind == OS_EventKind_Release && ev->key != OS_Key_RightMouseButton) ||
       (ev->kind == OS_EventKind_Release && ev->key != OS_Key_MiddleMouseButton) ||
       (ev->kind == OS_EventKind_Scroll))
  {
    os_eat_event(&leftover_events, ev);
  }
}
```

All press and release events will be eaten regardless of `ev->key`.

**Comment by ryanfleury** [comment]
Oops... must've been on autopilot there. Fixed again in baca24c0377d9498fe6830ac1e16d9da8b7607e0.

**Comment by ryanfleury** [comment]
I'm experiencing issues on my end with unconsumed events sticking around, so I am going to disable this behavior & reopen this issue until this is fully fixed.

**Comment by aswerw** [comment]
It seems fixed for the menu items and tabs, but the text view widget seems to ignore the down event now as well (text cursor doesn't update).

**Comment by ryanfleury** [comment]
Can you show a video of what you're seeing? I am not seeing that on my end.

**Comment by ryanfleury** [comment]
I think I see one potential issue - you might be expecting whatever mouse positions are attached to the mouse press/release events (presumably auto-generated) to show up in the selection itself, without forcibly changing what the mouse position is?

If that's the case, can you expand on the intended use case you have in mind for that?

**Comment by aswerw** [comment]
https://github.com/EpicGames/raddebugger/assets/10407330/ccda40d1-5f02-47c4-a8b3-82ac27934e80

Yeah, the fast click events are auto-generated to reproduce it consistently for video.

```
#pragma comment(lib, "user32.lib")

#include <Windows.h>

int main()
{
    if(RegisterHotKey(0, 1, MOD_NOREPEAT, 'A'))
    {
        MSG Msg = {0};
        while(GetMessage(&Msg, 0, 0, 0) != 0)
        {
            if(Msg.message == WM_HOTKEY)
            {
                int const Count = 2;
                INPUT Inputs[Count] = {};

                Inputs[0].type = INPUT_MOUSE;
                Inputs[0].mi.dwFlags = MOUSEEVENTF_LEFTDOWN;

                Inputs[1].type = INPUT_MOUSE;
                Inputs[1].mi.dwFlags = MOUSEEVENTF_LEFTUP;

                if(SendInput(Count, Inputs, sizeof(INPUT)) != Count)
                {
                    break;
                }
            }
        } 
    }

    return 0;
}
```
(Also tested with a Logitech mouse macro, and my actual use case, remote desktop.)

**Comment by ryanfleury** [comment]
Ahhhhh okay makes perfect sense, thanks. Looks like I've fixed this on my end in 77337f9e860cd32c473878932fded8c56e5e355a. Can you give that a try?

**Comment by aswerw** [comment]
Seems to be working! 👍
